### DISCUSSION TITLE
i want to purge all revisions of keys from etcd db daily, what should i use in auto compaction?

### DISCUSSION THREAD

**Discussion by ssengar** [etcd-io/etcd#16363] [Q&A] [answered]
i want to purge all older revisions of keys from etcd db daily only keeping the current revision, what should pass in etcd startup flags to achieve defrag i'm running as a cronjob.

**Comment by jmhbnz** [comment, marked as answer]
Hi @ssengar - Thanks for your question. The flags you will want are `--auto-compaction-retention` and `--auto-compaction-mode`.

Example for daily auto compaction: `etcd --auto-compaction-mode=periodic --auto-compaction-retention=24h`

Refer: 
 - https://etcd.io/docs/v3.5/op-guide/maintenance/#auto-compaction
 - https://etcd.io/docs/v3.5/op-guide/configuration

**Reply by ssengar** [reply]
Hi @jmhbnz ,

thanks for your reply , just to understand these flags if i put --auto-compaction-mode=periodic --auto-compaction-retention=24h apart from latest revision all other revisions will be dropped every 24hrs...is my understanding correct?

**Reply by jmhbnz** [reply]
That's my understanding yes. There is a bit of nuance about the sliding window approach etcd uses that I believe runs every 1/6th of an hour to track revisions of changed records but the actual compaction process should be every 24h with those flags and be pruning any redundant versions yes.
